### ISSUE TITLE
Error with installing requirements.txt during deployment on tensorflow instance

### ISSUE THREAD

**Issue by vincentzhang** [aws/sagemaker-python-sdk#399]
### System Information
- **Framework (e.g. TensorFlow) / Algorithm (e.g. KMeans)**: Tensorflow / Custom Algorithm
- **Framework Version**: 1.9
- **Python Version**: Python 2.7
- **CPU or GPU**: GPU & CPU
- **Python SDK Version**: Don't know how to find it
- **Are you using a custom image**: No

### Describe the problem
I have a model artifact on Sagemaker that I want to deploy. 
It gets an error related to not being able to install from requirements.txt:

`CalledProcessError: Command '['pip', 'install', '-r', '/opt/ml/code/requirements.txt']' returned non-zero exit status 2 `

My requirements.txt file contains one line only:
scikit-image

### Minimal repro / logs
The issue is apparently that I cannot properly install python packages through requirements.txt on the endpoint docker instance.  I've tried to use the estimator api, but it didn't work. 
```
from sagemaker.tensorflow import TensorFlow
estimator = TensorFlow(...,  source_dir='./', requirements_file='requirements.txt', ... )
estimator.fit(inputs)
predictor = estimator.deploy(initial_instance_count=1,
                                   instance_type='ml.p2.8xlarge')
```
That's why I tried the following, but it still gave the same error.

- **Exact command to reproduce**:
```
from sagemaker.tensorflow.model import TensorFlowModel
sagemaker_model = TensorFlowModel(model_data = 's3://xxxx/model.tar.gz',
                                  role = role,
                                  source_dir='.',
                                  env={'SAGEMAKER_REQUIREMENTS': 'requirements.txt'},
                                  entry_point='train_deploy_sagemaker.py',
                                  framework_version='1.9')

predictor = sagemaker_model.deploy(initial_instance_count=1,
                                   instance_type='ml.p2.8xlarge')
```

It seems to be related to the following: #182  #91 
 Any help is appreciated.

**Comment by mvsusp** [comment]
Hello @vincentzhang,

I was able to reproduce your issue. It happened because multiple workers were trying to install the dependencies at the same time creating deadlocking issues with pip. I submitted this PR https://github.com/aws/sagemaker-containers/pull/109 with a fix for the issue.

Thanks for using SageMaker.

**Comment by vincentzhang** [comment]
@mvsusp awesome. Thanks for fixing it! Looking forward to the merging of this PR

**Comment by ChoiByungWook** [comment]
@vincentzhang,

Looks like the fix has been merged and released.

https://github.com/aws/sagemaker-containers/releases/tag/v1.0.7

**Comment by vincentzhang** [comment]
I've found out the version from the cloudwatch logs: 
sagemaker-container-support==1.0.5
sagemaker-tensorflow==1.9.0.1.0.5
sagemaker-tensorflow-container==1.0.0

But it would still be good to know if there's any other way to find out and update the version. Also, it would be great to know the estimate of the release time to production if possible. 
Btw, I tried to update the sagemaker on my notebook instance but it didn't make any difference.
